src/dolphin/downloadDolphin.ts

```diff
@@ -145,100 +146,97 @@ async function installDolphin(
-            }
-          });
-        }
-      });
-      const binaryLocation = path.join(dolphinPath, "Slippi Dolphin.app", "Contents", "MacOS", "Slippi Dolphin");
-      const userInfo = os.userInfo();
-      try {
-        await fs.chmod(path.join(dolphinPath, "Slippi Dolphin.app"), "777");
-        await fs.chown(path.join(dolphinPath, "Slippi Dolphin.app"), userInfo.uid, userInfo.gid);
-        await fs.chmod(binaryLocation, "777");
-        await fs.chown(binaryLocation, userInfo.uid, userInfo.gid);
-      } catch (err) {
-        log(`could not chmod/chown ${type} Dolphin\n${err}`);
-      }
-
-      // Move backed up User folder and user.json
-      if (alreadyInstalled) {
-        const oldUserFolder = path.join(backupLocation, "Slippi Dolphin.app", "Contents", "Resources", "User");
-        const newUserFolder = path.join(dolphinResourcesPath, "User");
-
-        if (await fs.pathExists(oldUserFolder)) {
-          log("moving User folder...");
-          await fs.move(oldUserFolder, newUserFolder);
-        } else {
-          log("no old User folder to move");
-        }
-        await fs.remove(backupLocation);
-      }
+      await installDolphinOnMac(assetPath, dolphinPath, log);
       break;
     }
     case "linux": {
-      // Delete the existing app image if there is one
-      try {
-        const dolphinAppImagePath = await findDolphinExecutable(type);
-        // Delete the existing app image if it already exists
-        if (await fileExists(dolphinAppImagePath)) {
-          log(`${dolphinAppImagePath} already exists. Deleting...`);
-          await fs.remove(dolphinAppImagePath);
-        }
-      } catch (err) {
-        log("No existing AppImage found");
-      }
-
-      if (cleanInstall) {
-        const userFolder = type === DolphinLaunchType.NETPLAY ? "SlippiOnline" : "SlippiPlayback";
-        await fs.remove(path.join(app.getPath("home"), ".config", userFolder));
-      }
-
-      const zip = new AdmZip(assetPath);
-      zip.extractAllTo(dolphinPath, true);
-
-      // Make the file executable
-      const dolphinAppImagePath = await findDolphinExecutable(type);
-      log(`Setting executable permissions...`);
-      await fs.chmod(dolphinAppImagePath, "755");
+      await installDolphinOnLinux(type, assetPath, dolphinPath, log);
       break;
     }
     default: {
       throw new Error(`Installing Netplay is not supported on this platform: ${process.platform}`);
     }
   }
 }
+
+async function installDolphinOnMac(assetPath: string, dolphinPath: string, log: (message: string) => void) {
+  const backupLocation = dolphinPath + "_old";
+  const dolphinResourcesPath = path.join(dolphinPath, "Slippi Dolphin.app", "Contents", "Resources");
+
+  const alreadyInstalled = await fs.pathExists(dolphinResourcesPath);
+  let shouldRestore = false;
+  if (alreadyInstalled) {
+    log(`${dolphinResourcesPath} already exists. Moving...`);
+    await fs.move(dolphinPath, backupLocation);
+    shouldRestore = true;
```

Comment:
Why do we track `shouldRestore`? If `alreadyInstalled` is true, then wouldn't `shouldRestore` also be guaranteed to be true?